src/controller/CHIPDeviceController.cpp

```diff
@@ -1891,21 +1646,20 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
             mCommissioningDelegate->CommissioningStepFinished(CHIP_ERROR_INVALID_ARGUMENT, report);
             return;
         }
-        ChipLogProgress(Controller, "Enabling thread network");
-        NetworkCommissioningCluster netCom;
-        netCom.Associate(proxy, 0);
-        netCom.ConnectNetwork(mSuccess.Cancel(), mFailure.Cancel(), extendedPanId, breadcrumb);
+
+        NetworkCommissioning::Commands::ConnectNetwork::Type request;
+        request.networkID  = extendedPanId;
+        request.breadcrumb = breadcrumb;
+        StopSessionIf(SendCommand<NetworkCommissioningCluster>(proxy, request, OnSuccess<decltype(request)::ResponseType>));
     }
     break;
     case CommissioningStage::kFindOperational: {
         UpdateDevice(proxy->GetDeviceId());
     }
     break;
     case CommissioningStage::kSendComplete: {
-        ChipLogProgress(Controller, "Calling commissioning complete");
-        GeneralCommissioningCluster genCom;
-        genCom.Associate(proxy, 0);
-        genCom.CommissioningComplete(mSuccess.Cancel(), mFailure.Cancel());
+        GeneralCommissioning::Commands::CommissioningComplete::Type request;
+        StopSessionIf(SendCommand<NetworkCommissioningCluster>(proxy, request, OnSuccess<decltype(request)::ResponseType>));
```

Comment:
I suspect CommissioningComplete and ArmFailSafe invocations will now fail.  As far as I know, accessories aren't returning correct responses for these.

See https://github.com/project-chip/connectedhomeip/issues/13327.

The legacy cluster code seemed to tolerate this.  I don't believe the new IM code will.

My solution in the #13294 feature branch was to simply tolerate non-success for these two invocaitons.

Reply:
You "were" right until https://github.com/project-chip/connectedhomeip/commit/5cd7eee9810bc8c06a56ad3a2a5b1b57478a5ac0 ;)

Reply:
Ah, that's great.  Thank you.

Reply:
Yes, tried it myself.  The fix works great.  Thanks again.